### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by KellyAH**
- added [minitest](https://github.com/seattlerb/minitest) unit tests for the task and thought models
NOTE: we'll use minitest for now and either add or swap to Rspec tests if needed
- fixed auto-generated controller tests, because redirects were changed
- updated readme with how to run tests

# reference docs we used
[Good overview](https://guides.rubyonrails.org/v3.2/testing.html) of the kinds of unit tests. (latest rails tests [doc](https://www.simform.com/unit-testing-vs-functional-testing/))
https://www.simform.com/unit-testing-vs-functional-testing/
[Launch School unit tests doc](https://launchschool.com/blog/assert-yourself-an-introduction-to-minitest)

[mini tests vs rspec](https://stackoverflow.com/questions/12470601/minitest-and-rspec)
[exists? method](https://apidock.com/rails/ActiveRecord/Base/exists%3f/class)
[rails Model_name.new vs  Model_name.create](https://stackoverflow.com/questions/2472393/rails-new-vs-create)
[unit test vs integration test](https://medium.com/android-testing-daily/unit-test-vs-integration-tes-fba13b92fbf6)

### CODE REVIEW

**Review by karolina-benitez — APPROVED** [review]
LGTM!

**Review by NOI-lgtm — APPROVED** [review]
👍 💯

File: test/models/task_test.rb [hunk 1]
```diff
@@ -1,7 +1,33 @@
...
+  end
+
+  def test_tasks_not_created_when_missing_priorty
+    record_created = Task.new(description: 'blah').save
+    assert(record_created == false)
+  end
+
+  def test_tasks_not_created_when_missing_description
+    record_created = Task.new(priority: 'low').save
+    assert(record_created == false)
```
Comment (KellyAH) [inline test/models/task_test.rb:26]: refute(record_created)

File: test/models/task_test.rb [hunk 2]
```diff
@@ -1,7 +1,33 @@
...
   # end
 end
+
+class TaskModelTest < Minitest::Test
+  # Testing Task.new and not Task.create because Task.new is used the TasksController.
+  def test_task_created_when_required_data_exists
+    record_created = Task.new(description: 'blahfoobar', priority: 'low').save
+    assert(record_created == true)
+    fetch_created_record = Task.where(description: 'blahfoobar', priority: 'low').exists?
+    assert(fetch_created_record == true)
```
Comment (KellyAH) [inline test/models/task_test.rb:16]: assert_equals(fetch_created_record, true)
assert(fetch_created_record)

**Pull request merged**